mitiq/shadows/classical_postprocessing.py

```diff
@@ -343,7 +343,7 @@ def expectation_estimation_shadow(
             np.nonzero(np.all(u_lists_shadow[idxes] == target_obs, axis=1))[0]
```

Comment:
Could you assign `np.nonzero(np.all(u_lists_shadow[idxes] == target_obs, axis=1)` to a new variable? 
I think it would make the code easier to understand, especially since the same long expression is used twice.